src/ExpressionParser.php

```diff
@@ -579,6 +579,11 @@ public function parseArguments($namedArguments = false, $definition = false, $al
         while (!$stream->test(/* Token::PUNCTUATION_TYPE */ 9, ')')) {
             if (!empty($args)) {
                 $stream->expect(/* Token::PUNCTUATION_TYPE */ 9, ',', 'Arguments must be separated by a comma');
+
+                // if the comma above was a trailing comma, early exit the argument parse loop
+                if ($stream->test(/* Token::PUNCTUATION_TYPE */ 9, ')')) {
+                    break;
+                }
```

Comment:
The nice part of this addition here, is that it prevents usage with no arguments.

So `_self.test(a,)` is allowed as macro call, `_self.test(,)` isn't however.